internal/powerplatform/services/data_record/resource_data_record.go

```diff
@@ -0,0 +1,338 @@
+		resp.Diagnostics.AddError(fmt.Sprintf("Client error when deleting %s_%s", r.ProviderTypeName, r.TypeName), err.Error())
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("DELETE RESOURCE END: %s", r.ProviderTypeName))
+}
+
+func (r *DataRecordResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func convertColumnsToState(ctx context.Context, apiClient *DataRecordClient, currentState *DataRecordResourceModel, environmentId string, tableLogicalName string, columns map[string]interface{}) *DataRecordResourceModel {
+	var objectType = map[string]attr.Type{
+		"table_logical_name": types.StringType,
+		"data_record_id":     types.StringType,
+	}
+
+	var old_columns map[string]interface{}
+	jsonColumns, _ := json.Marshal(currentState.Columns.String())
+	unquotedJsonColumns, _ := strconv.Unquote(string(jsonColumns))
+	json.Unmarshal([]byte(unquotedJsonColumns), &old_columns)
+
+	attributeTypes := make(map[string]attr.Type)
+	attributes := make(map[string]attr.Value)
+
+	for key, value := range old_columns {
+		switch value.(type) {
+		case bool:
+			v, ok := columns[key].(bool)
+			if ok {
+				attributeTypes[key] = types.BoolType
+				attributes[key] = types.BoolValue(v)
+			}
+		case int64:
+			v, ok := columns[key].(int64)
+			if ok {
+				attributeTypes[key] = types.Int64Type
+				attributes[key] = types.Int64Value(v)
+			}
+		case float64:
+			v, ok := columns[key].(float64)
+			if ok {
+				attributeTypes[key] = types.Float64Type
+				attributes[key] = types.Float64Value(v)
+			}
+		case string:
+			v, ok := columns[key].(string)
+			if ok {
+				attributeTypes[key] = types.StringType
+				attributes[key] = types.StringValue(v)
+			}
+		case map[string]interface{}:
+			v, ok := columns[fmt.Sprintf("_%s_value", key)].(string)
+			if ok {
+				entityLogicalName := apiClient.GetEntityRelationDefinitionInfo(ctx, environmentId, tableLogicalName, key)
+				dataRecordId := v
+
+				nestedObjectType := types.ObjectType{
+					AttrTypes: objectType,
+				}
+				nestedObjectValue, _ := types.ObjectValue(
+					objectType,
+					map[string]attr.Value{
+						"table_logical_name": types.StringValue(entityLogicalName),
+						"data_record_id":     types.StringValue(dataRecordId),
+					},
+				)
+
+				attributeTypes[key] = nestedObjectType
+				attributes[key] = nestedObjectValue
+			}
+		case []interface{}:
+			setObjectValues := []attr.Value{}
+			var setObjectType = types.ObjectType{
+				AttrTypes: objectType,
+			}
+			relationMap, _ := apiClient.GetRelationData(ctx, currentState.Id.ValueString(), environmentId, tableLogicalName, key)
+
+			for _, rawItem := range relationMap {
+				item := rawItem.(map[string]interface{})
+
+				relationTableLogicalName := apiClient.GetEntityRelationDefinitionInfo(ctx, environmentId, tableLogicalName, key)
+				dataRecordId := ""
+
+				for itemKey, itemValue := range item {
```

Comment:
w discussed that, its not how recordid should be retrieved, please go to metadata and fild "primaryfielname" attribute value

Reply:
done 👍